src/bptl/work_units/email/serializers.py

```diff
@@ -0,0 +1,37 @@
+from django.utils.translation import ugettext_lazy as _
+
+from djchoices import ChoiceItem, DjangoChoices
+from rest_framework import serializers
+
+
+class ValidTemplateChoices(DjangoChoices):
+    generiek = ChoiceItem("generiek", "../templates/generic_email.txt")
+    accordering = ChoiceItem("accordering", "../templates/review.txt")
+    advies = ChoiceItem("advies", "../templates/review.txt")
```

Comment:
relative paths look weird, especially because django has template resolving already based on app name.

`django.template.loader.get_template` will search according to the config in the `TEMPLATES` setting.

Especially within a docker context where you can mount volumes containing custom templates, this is relevant for broader adoption than just the templates in this codebase.

I'd change this into `email/mails/generic.txt` and `email/mails/review.txt` rather.

I'm also not sure that using `DjangoChoices` here makes sense - because you're using the same label for different values which makes it impossible in user interfaces to select the correct choice.

Shouldn't this rather be a simple mapping?

```suggestion
MAIL_TEMPLATES = {
    "generiek": "email/mails/generic.txt",
    "accordering": "email/mails/review.txt",
    "advies": "email/mails/review.txt",
}
```

and then the serializer fields becomes something rather like:

```py
template = serializers.ChoiceField(choices=((key, key) for key in MAIL_TEMPLATES)
```

Reply:
fair enough